pkg/binlog/event/common.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package event
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/pingcap/errors"
+	gmysql "github.com/siddontang/go-mysql/mysql"
+	"github.com/siddontang/go-mysql/replication"
+
+	"github.com/pingcap/dm/pkg/gtid"
+)
+
+// DDLDMLResult represents a binlog event result for generated DDL/DML.
+type DDLDMLResult struct {
+	Events     []*replication.BinlogEvent
+	Data       []byte // data contain all events
+	LatestPos  uint32
+	LatestGTID gtid.Set
+}
+
+// GenCommonFileHeader generates a common binlog file header.
+// for MySQL:
+//   1. BinLogFileHeader, [ fe `bin` ]
+//   2. FormatDescriptionEvent
+//   3. PreviousGTIDsEvent
+// for MariaDB:
+//   1. BinLogFileHeader, [ fe `bin` ]
+//   2. FormatDescriptionEvent
+//   3. MariadbGTIDListEvent
+//   -. MariadbBinlogCheckPointEvent, not added yet
+func GenCommonFileHeader(flavor string, serverID uint32, gSet gtid.Set) ([]*replication.BinlogEvent, []byte, error) {
```

Comment:
for `Genxxx` functions,  I suggest to add `latestPos` for all them

Reply:
the header has no meaningful pos/latestPos
1. 4 bytes magic num
2. (123-4) bytes FormatDescriptionEvent (currently with CRC32 checksum enabled)
3. variable length PreviousGTIDsEvent/MariadbGTIDListEvent

Reply:
while I saw this code, I had to confirm the position mechanism, although it is correct, but not uniform

Reply:
yeh, but if add a `latestPos`, then do not know which value should be passed.:worried:

Reply:
we would use like this way
```
ev = GenXX(latestPos)
updateLatePos(ev)
```
so it's not a problem, expect diving into it

but it's not a big problem

Reply:
check latest GTID whether is one of the last GTID in previous GTIDs?

Reply:
yep